package/MDAnalysis/core/selection.py

```diff
@@ -226,7 +226,7 @@ class ByResSelection(UnarySelection):
 
     def apply(self, group):
         res = self.sel.apply(group)
-        unique_res = unique_int_1d(res.resindices.astype(np.int64))
+        unique_res = unique_int_1d(res.resindices.astype(np.intp))
```

Comment:
Is this `astype` necessary? I think it required coercion earlier as `resindices` are `intp` typically.

Reply:
I've pushed in a commit to remove it--32-bit Windows suite passed locally without it. I didn't have time to run the suite for 64-bit linux as well, but fingers crossed.